src/async_graphql_hyper.rs

```diff
@@ -126,6 +126,23 @@ impl GraphQLResponse {
     Ok(response)
   }
 
+  pub fn to_response_with_status_code(self, status_code: StatusCode) -> Result<Response<hyper::Body>> {
```

Comment:
@meskill how about migrating from the function above to the new function instead?

And I added that function because I was asked asked to do so [here](https://github.com/tailcallhq/tailcall/pull/673#discussion_r1404735381)